spec/ssh_scan/rule_engine_spec.rb

```diff
@@ -0,0 +1,125 @@
+require 'rspec'
+require 'ssh_scan/rule_engine'
+
+# ignore absurdities in rules, they have been created to test the rule engine
+describe SSHScan::RuleEngine do
+  context "match single rule" do
+    it "eq should work for non numerics" do
+      result = {:ssh_lib => "openssh"}
+      expect(SSHScan::RuleEngine.matchSingleRule("ssh_lib_eq", "openssh", result)).to be true
+      expect(SSHScan::RuleEngine.matchSingleRule("ssh_lib_eq", "libssh", result)).to be false
+    end
+
+    it "neq should work for non numerics" do
+      result = {:ssh_lib => "openssh"}
+      expect(SSHScan::RuleEngine.matchSingleRule("ssh_lib_neq", "dropbear", result)).to be true
+      expect(SSHScan::RuleEngine.matchSingleRule("ssh_lib_neq", "openssh", result)).to be false
+    end
+
+    it "eq should work for numerics" do
+      result = {:ssh_version => 2.0}
+      expect(SSHScan::RuleEngine.matchSingleRule("ssh_version_eq", 2.0, result)).to be true
+      expect(SSHScan::RuleEngine.matchSingleRule("ssh_version_eq", 1.99, result)).to be false
+    end
+
+    it "neq should work for numerics" do
+      result = {:ssh_version => 2.0}
+      expect(SSHScan::RuleEngine.matchSingleRule("ssh_version_neq", 1.99, result)).to be true
+      expect(SSHScan::RuleEngine.matchSingleRule("ssh_version_neq", 2.0, result)).to be false
+    end
+
+    it "gt should work for numerics" do
+      result = {:ssh_version => 2.0}
+      expect(SSHScan::RuleEngine.matchSingleRule("ssh_version_gt", 1.3, result)).to be true
+      expect(SSHScan::RuleEngine.matchSingleRule("ssh_version_gt", 2.0, result)).to be false
```

Comment:
Recommend using the following cases:

expect(SSHScan::RuleEngine.matchSingleRule("ssh_version_gt", 1.99, result)).to be true
expect(SSHScan::RuleEngine.matchSingleRule("ssh_version_gt", 2.0, result)).to be false
expect(SSHScan::RuleEngine.matchSingleRule("ssh_version_gt", 2.1, result)).to be false